### PR TITLE
ADD Florida Supplement Courts

### LINKED ISSUE #1733: Florida Supplement Courts
To add the first Florida Supplement case law we will need to add the following courts no in our system.

        # Bay County Circuit Court
        # Hendry County Circuit Court
        # Duval County Circuit Court
        # Palm Beach County Circuit Court
        # Palm Beach County Criminal Court of Record
        # Walton County Circuit Court
        # Broward County Small Claims Court
        # Flagler County Circuit Court
        # Dade County Court
        # Pinellas County Court
        # Florida Railroad & Public Utilities Commission
        # Orange County Judge's Court
        # Nassau County Circuit Court
        # Lee County Judge's Court
        # Escambia County Circuit Court
        # Dade County Metropolitan Court
        # Alachua County Circuit Court
        # Volusia County Judge's Court
        # Suwannee County Judge's Court
        # Okeechobee County Circuit Court
        # Indian River County Circuit Court
        # Suwanee County Circuit Court
        # Palm Beach County Small Claims Court
        # Florida Public Utilities Commission
        # Dade County Small Claims Court
        # Dade County Judges' Court
        # Lake County Judge's Court
        # Osceola County Circuit Court
        # Duval County Civil Court of Record
        # Columbia County Circuit Court
        # Florida Public Service Commission
        # Broward County Court
        # Sumter County Circuit Court
        # Orange County Circuit Court
        # Bradford County Circuit Court
        # Calhoun County Circuit Court
        # Highlands County Circuit Court
        # Pinellas County Civil Court of Record
        # De Soto County Circuit Court
        # Manatee County Circuit Court
        # Hillsborough County Court
        # Miami Beach Municipal Court
        # Pinellas County Civil and Criminal Court of Record
        # St. Lucie County Small Claims Court
        # Industrial Commission
        # Charlotte County Judge's Court
        # Pasco County Circuit Court
        # Dade County Judge's Court
        # Hamilton County Judges' Court
        # Escambia County Court of Record
        # Pinellas County Circuit Court
        # Santa Rosa County Circuit Court
        # Lee County Circuit Court
        # Levy County Circuit Court
        # Palm Beach County Court
        # Gadsden County Circuit Court
        # Duval County Ciruit Court
        # Broward County Circuit Court
        # Leon County Circuit Court
        # Orange County Court
        # Collier County Circuit Court
        # Broward County Judges' Court
        # Pinellas County Judge's Court
        # Polk County Circuit Court
        # Surfside Municipal Court
        # Railroad & Public Utilities Commission
        # Monroe County Court
        # Hamilton County Circuit Court
        # Palm Beach County Juvenile and Domestic Relations Court
        # Dade County Circuit Court
        # Martin County Circuit Court
        # Munroe County Circuit Court
        # Sarasota County Circuit Court
        # St. Lucie County Circuit Court
        # Brevard County Court of Record
        # Marion County Circuit Court
        # Monroe County Circuit Court
        # Volusia County Circuit Court
        # Monroe County Judge Court
        # Brevard County Court
        # Florida Industrial Commission
        # Florida County Court
        # Dade County Civil Court
        # Polk County Court
        # Charlotte County Circuit Court
        # Hillsborough County Circuit Court
        # Palm Beach County Judge’s Court
        # Florida Circuit Court
        # Miami Municipal Court
        # Dade County Civil Court of Record
        # Lake County Circuit Court
        # Florida Real Estate Commission
        # Seminole County Circuit Court
        # Duval County Judge's Court
        # Union County Circuit Court
        # Okaloosa County Circuit Court
        # Broward County Court Record
        # Glades County Circuit Court
        # Clay County Circuit Court
        # Sarasota County Judge's Court
        # Brevard County Circuit Court
        # Citrus County Circuit Court

This doesn't include the second Florida Supplement.  

So - most are entry courts but there are at least some commissions that I've never heard of.

### PR DESCRIPTION
**Pull request by flooie**
PR Fixes #1733 

PR required an updated courts-db version.  
Adds a fixture for Florida supplemental courts and divides courts into groupings. 

Adds test to show Courts-DB was updated and the new court exists and is correct.

### CODE REVIEW

**Comment by flooie** [comment]
@mlissner 

I had to modify the docker file slightly to get it to build.  It kept throwing an error related to python-igraph.  Looks like that module can be somewhat finicky and you can find lots of threads about it not installing or needing to be built. 

Other than that its a very small update.

File: cl/corpus_importer/tests.py [hunk 1]
```diff
@@ -716,3 +717,9 @@ def test_short_opinion_matching(self) -> None:
 
         bad_match = compare_documents(harvard_characters, bad_characters)
         self.assertEqual(bad_match, 81)
+
+    def test_florida_court_addition(self):
+        """Can we find a court from our new fixture?"""
+
+        result = courts_db.find_court("Broward County Judges' Court")
+        self.assertEqual(result[0], "flajudct9", msg="Missing Florida Court")
```
Comment (mlissner) [inline cl/corpus_importer/tests.py:725]: Will this be a useful test in a week? Is it worth having in our codebase long term? Is there a better test that'd ensure that our code is working properly? It seems like it's just testing that a dependency is upgraded?

File: cl/search/fixtures/florida_state_court_data.json [hunk 2]
```diff
@@ -0,0 +1,3726 @@
+[
+  {
+    "fields": {
+      "jurisdiction": "ST",
+      "position": 355.05,
+      "short_name": "",
+      "full_name": "Florida Judge's Courts",
+      "citation_string": "",
```
Comment (mlissner) [inline cl/search/fixtures/florida_state_court_data.json:8]: I think `citation_string` is used in a few places and we'll need to include it. Can you check?

File: cl/search/fixtures/florida_state_court_data.json [hunk 3]
```diff
@@ -0,0 +1,3726 @@
+[
+  {
+    "fields": {
+      "jurisdiction": "ST",
+      "position": 355.05,
+      "short_name": "",
```
Comment (mlissner) [inline cl/search/fixtures/florida_state_court_data.json:6]: I don't know if `short_name` is used much. Regardless, I suppose we can just put a copy/paste of the full_name field into it if we don't want to go to the trouble of figuring it out for each court. Want to poke around and see how much the field is used?

Comment (mlissner) [reply cl/search/fixtures/florida_state_court_data.json:6]: Oh, I'm seeing at second glance that it's populated elsewhere. Let's make sure it's consistent.

File: cl/search/fixtures/florida_state_court_data.json [hunk 4]
```diff
@@ -0,0 +1,3726 @@
...
+      "full_name": "Florida Judge's Courts",
```
Comment (mlissner) [inline cl/search/fixtures/florida_state_court_data.json:7]: Is this apostrophe right?

File: cl/search/fixtures/florida_state_court_data.json [hunk 5]
```diff
@@ -0,0 +1,3726 @@
...
+      "notes": "Created as part of import of Case.Law data."
+    },
+    "model": "search.court",
+    "pk": "flajudct"
+  },
+  {
+    "fields": {
+      "jurisdiction": "ST",
+      "position": 355.0501,
+      "short_name": "Broward County Judges' Court",
```
Comment (mlissner) [inline cl/search/fixtures/florida_state_court_data.json:25]: Or is this apostrophe wrong? :)

File: cl/search/fixtures/florida_state_court_data.json [hunk 6]
```diff
@@ -0,0 +1,3726 @@
...
+    },
+    "model": "search.court",
+    "pk": "flajudct4"
+  },
+  {
+    "fields": {
+      "jurisdiction": "ST",
+      "position": 355.051,
+      "short_name": "Palm Beach County Judge\u00e2\u20ac\u2122s Court",
+      "full_name":  "Palm Beach County Judge\u00e2\u20ac\u2122s Court",
```
Comment (mlissner) [inline cl/search/fixtures/florida_state_court_data.json:197]: Yikes

**Review by mlissner — CHANGES_REQUESTED** [review]
Just a couple tweaks.

File: cl/corpus_importer/tests.py (hunk 1 above)
Comment (flooie) [reply cl/corpus_importer/tests.py]: Removed.

File: cl/search/fixtures/florida_state_court_data.json (hunk 2 above)
Comment (flooie) [reply cl/search/fixtures/florida_state_court_data.json]: Added citation strings for all.

File: cl/search/fixtures/florida_state_court_data.json (hunk 5 above)
Comment (flooie) [reply cl/search/fixtures/florida_state_court_data.json]: This one is wrong and they have been made uniform

File: cl/search/fixtures/florida_state_court_data.json (hunk 6 above)
Comment (flooie) [reply cl/search/fixtures/florida_state_court_data.json]: Fixed

File: cl/search/fixtures/florida_state_court_data.json (hunk 4 above)
Comment (flooie) [reply cl/search/fixtures/florida_state_court_data.json:7]: Yes this one is correct.

**Pull request merged**